### ISSUE TITLE
[Feature Request] AMD FidelityFX Super Resolution (FSR) and/or other neuro rescalers?

### ISSUE THREAD

**Issue by DartPower** [PCSX2/pcsx2#4649]
What about implementing AMD FidelityFX Super Resolution (FSR) and/or other neuro rescalers?

https://github.com/GPUOpen-Effects/FidelityFX-FSR
https://github.com/igv/FSRCNN-TensorFlow
https://github.com/bloc97/Anime4K

Example: (My idea from here)
https://github.com/7Brandyn7/Magpie

**Comment by refractionpcsx2** [comment]
You're about the 10 millionth person to request/ask about it, congratulations! I will mail you your prize directly! :)

But seriously, we're already looking in to it/working on it, so it'll come if and when we deem it ready.

**Comment by githubribbon45** [comment]
Now that PCSX2 supports a wide range of APIs, I would be curious to know in which state the FidelityFX 1.0 feature is in this project.

We have seen how it greatly improves RPCS3's emulation performance while preserving good image quality at high resolution scaling.

+ In the case of PCSX2, there is a great opportunity to takes advantage from, upscaling IPU movies or even the software render with such a technology.

**Comment by dio-gh** [comment]
Months ago there was an internal test build with FSR 1.0, but due to various PS2 related factors (mainly it needing to be implemented before AA), it's visual performance was subpar. It was deemed not good enough to warrant a public release.

There's been no progress since.

Somebody please also change the ticket's title by the way, two of those three listed methods are perfectly vanilla scalers without the slightest hint of neural networks of any kind.

**Comment by dio-gh** [comment]
Great idea for another meta ticket.

**Comment by lightningterror** [comment]
This was tested a while back and it didn't provide much benefit, closing as trivial.

**Comment by boludoz** [comment]
> 

The point is that it improves the graphic quality

**Comment by refractionpcsx2** [comment]
Thanks for your valuable insight.

We already have CAS, we tried FSR, it did nothing.

**Comment by AXVIRE** [comment]
I used the PS2 core from Retroarch at base resolution and it looks amazing with the FSR filter, I don't understand why they say it doesn't work. I also use Lossless Scaling with PCSX2 and it also improves the image quality, just that you have to calculate the scaling factor manually which is annoying, but it's worth it.

**Comment by JordanTheToaster** [comment]
> looks amazing with the FSR filter

FSR is two parts the upscaler and CAS we already have CAS which does improve the quality of native resolution output.

**Comment by AXVIRE** [comment]
> > looks amazing with the FSR filter
> 
> FSR is two parts the upscaler and CAS we already have CAS which does improve the quality of native resolution output.
> 
> 

But what we want is EASU, the upscaling part, I use FSR without RCAS (sharpness at 0) in Lossless Scaling, I guess I'll have to use that program.
What I notice CAS does is only sharpen the textures, it does not improve the resolution.
